src/core/stream_recv.c

```diff
@@ -807,6 +807,19 @@ QuicStreamRecvFlush(
             Event.RECEIVE.Flags |= QUIC_RECEIVE_FLAG_FIN; // TODO - 0-RTT flag?
         }
 
+        if (Stream->ReceiveCompleteOperation == NULL) {
```

Comment:
I have a gut feeling this needs to be an interlocked operation, or at least a read that forces a memory barrier. But maybe its fine. Just worrying since its written from another thread, and the other side is an atomic.

Reply:
I'm pretty sure it's fine. This logic **always** runs before the other side. This is before we even indicate the receive event to the app, only after which may they call the receive complete; either on the same thread or after queuing work to another thread, which would have its own memory barrier in the process. Ok?